### PR TITLE
merge main into eslint branch

### PR DESCRIPTION
**Pull request by matthieu-crouzet**

### CODE REVIEW

File: packages/@o3r/design/builders/generate-style/helpers/style-renderer-options.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+  /** Sorting strategy of variables based on selected language */
+  const tokenListTransforms = ((language) => {
+    switch (language) {
+      case 'scss':
+      case 'sass': {
+        return [getTokenSorterByName, getTokenSorterByRef];
+      }
+      default: {
+        return [getTokenSorterByName];
```
Comment (kpanot) [inline packages/@o3r/design/builders/generate-style/helpers/style-renderer-options.ts:130]: missing new customFilter

File: eslint.shared.config.mjs [hunk 2]
```diff
@@ -138,7 +138,7 @@ export default [
       '@o3r/json-dependency-versions-harmonize': [
         'error',
         {
-          ignoredPackages: ['@o3r/build-helpers'],
+          ignoredPackages: ['@o3r/build-helpers', '@o3r/workspace-helpers'],
```
Comment (kpanot) [inline eslint.shared.config.mjs:141]: Not needed here, `@o3r/workspace-helpers` should be used only at workspace level

File: nx.json [hunk 3]
```diff
@@ -17,100 +17,126 @@
...
-      "default",
+      "base",
       "{projectRoot}/eslint*.config.mjs",
       "{projectRoot}/**/*.ts",
       "{projectRoot}/**/*.cts",
       "{projectRoot}/**/*.mts",
       "{projectRoot}/src/**/*.js",
       "{projectRoot}/src/**/*.mjs",
       "{projectRoot}/src/**/*.cjs",
+      "{projectRoot}/src/**/*.html",
```
Comment (kpanot) [inline nx.json:106]: missing also `{workspaceRoot}/package.json` and `{workspaceRoot}/yarnrc.yml`

File: tools/github-actions/get-npm-tag/package.json [hunk 4]
```diff
@@ -44,6 +44,7 @@
     "eslint-plugin-unused-imports": "^4.1.4",
     "globals": "^15.9.0",
     "jsonc-eslint-parser": "~2.4.0",
+    "tslib": "^2.6.2",
```
Comment (kpanot) [inline tools/github-actions/get-npm-tag/package.json:47]: surprising to see it here, should not be in `dependencies`?

File: tools/@o3r/build-helpers/project.json [hunk 5]
```diff
@@ -3,10 +3,14 @@
   "$schema": "https://raw.githubusercontent.com/nrwl/nx/master/packages/nx/schemas/project-schema.json",
   "projectType": "library",
   "sourceRoot": "packages/@o3r/build-helpers/scripts",
+  "tags": [],
   "targets": {
     "lint": {
       "executor": "nx:run-commands"
+    },
+    "build": {
```
Comment (kpanot) [inline tools/@o3r/build-helpers/project.json:11]: Not sure to understand the purpose of thiis empty `build`

**Comment by matthieu-crouzet** [comment]
Merge will be done later

**Pull request closed without merging**